srcs/config_parse/node.hpp

```diff
@@ -0,0 +1,12 @@
+#ifndef NODE_HPP_
+#define NODE_HPP_
+#include <string>
+
+struct Node {
+	std::string token_;
+	int         token_type_;
+	Node(const std::string &token, int token_type);
+	~Node();
+};
```

Comment:
public しかない struct は、コンストラクタ / デストラクタなしでこんな感じで宣言と同時に作れるみたいなので、これで十分かもしれないです
(struct は全部 public なので、変数名の `_` 終わりもいらなさそうです)
```cpp
const Node  token = {symbol_str, token_type};
```

```suggestion
struct Node {
	std::string token;
	int         token_type;
};
```
集成体初期化
https://ja.cppreference.com/w/cpp/language/aggregate_initialization